File: src/dependent/payments/payment_gateway_client.py
from decimal import Decimal


class PaymentGatewayClient:
    def charge(self, amount: Decimal, currency: str, reference: str) -> str:
        if amount is None or amount <= Decimal("0"):
            raise ValueError("Amount must be positive")
        if currency is None or currency == "":
            raise ValueError("Currency must be provided")
        return "pay_" + str(abs(hash(f"{amount}{currency}{reference}")))

File: src/dependent/payments/PaymentGatewayClient.java
package org.example.dependent.payments;

import java.math.BigDecimal;

public class PaymentGatewayClient {
    public String charge(BigDecimal amount, String currency, String reference) {
        if (amount == null || amount.compareTo(BigDecimal.ZERO) <= 0) {
            throw new IllegalArgumentException("Amount must be positive");
        }
        if (currency == null || currency.isEmpty()) {
            throw new IllegalArgumentException("Currency must be provided");
        }
        return "pay_" + Math.abs((amount.toPlainString() + currency + reference).hashCode());
    }
}


